File: src/main/java/infobip/test/shortener/account/OpenAccountCommand.java
package infobip.test.shortener.account;

import infobip.test.shortener.model.AccountData;
import org.axonframework.commandhandling.annotation.TargetAggregateIdentifier;
import org.immutables.value.Value;

@Value.Immutable
public interface OpenAccountCommand {

    @Value.Derived
    @TargetAggregateIdentifier
    default String getAccountId(){
        return getData().getAccountId();
    }

    AccountData getData();
    String getPassword();

}

File: src/main/java/infobip/test/shortener/account/OpenAccountCommand.kt
package infobip.test.shortener.account

import infobip.test.shortener.model.AccountData
import org.axonframework.commandhandling.annotation.TargetAggregateIdentifier

data class OpenAccountCommand(val data: AccountData? = null,
                              val password: String? = null) {

    @TargetAggregateIdentifier fun getAccountId() = data?.accountId
}